solidity/contracts/MailboxV2.sol

```diff
@@ -0,0 +1,224 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.8.0;
+
+// ============ Internal Imports ============
+import {Versioned} from "./upgrade/Versioned.sol";
+import {MerkleLib} from "./libs/Merkle.sol";
+import {MessageV2} from "./libs/MessageV2.sol";
+import {TypeCasts} from "./libs/TypeCasts.sol";
+import {IMessageRecipient} from "../interfaces/IMessageRecipient.sol";
+import {IInterchainSecurityModule, IUsesInterchainSecurityModule} from "../interfaces/IInterchainSecurityModule.sol";
+import {IMailboxV2} from "../interfaces/IMailboxV2.sol";
+
+// ============ External Imports ============
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import {ReentrancyGuardUpgradeable} from "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+
+contract MailboxV2 is
```

Comment:
Ultimately, no. The rationale for the suffixes for now was that such heavy modifications to these contracts would require creating an unreviewable PR.

For example, if I had named this contract `Mailbox` instead of `MailboxV2` I would have had to remove Outbox, Inbox, which would in turn have cascading effects on e.g. the SDK.

The V2 suffixes will be removed in a future PR but for now this seemed like the path that got us the most reviewable diffs.